### PR TITLE
e2e_windows: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by Songjoy**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?
```release-note
NONE

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/windows/host_process.go [hunk 1]
```diff
@@ -724,15 +724,9 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
-		framework.ExpectEqual(
-			strings.Contains(logs, "calling /healthz"),
-			true,
-			"app logs should contain 'calling /healthz'")
-
-		framework.ExpectEqual(
-			strings.Contains(logs, "status=failed"),
-			false,
-			"app logs should not contain 'status=failed")
+		gomega.Expect(strings.Contains(logs, "calling /healthz")).To(gomega.BeTrue(), "app logs should contain 'calling /healthz'")
```
Comment (carlory) [inline test/e2e/windows/host_process.go:727]:
```suggestion
		gomega.Expect(logs).To(gomega.ContainsSubstring("calling /healthz"), "app logs should contain 'calling /healthz'")
```

File: test/e2e/windows/host_process.go [hunk 2]
```diff
@@ -724,15 +724,9 @@ var _ = SIGDescribe("[Feature:WindowsHostProcessContainers] [MinimumKubeletVersi
...
+
+		gomega.Expect(strings.Contains(logs, "status=failed")).To(gomega.BeFalse(), "app logs should not contain 'status=failed'")
```
Comment (carlory) [inline test/e2e/windows/host_process.go:729]:
```suggestion
		gomega.Expect(logs).NotTo(gomega.ContainsSubstring("status=failed"), "app logs should not contain 'status=failed'")
```

File: test/e2e/windows/security_context.go [hunk 3]
```diff
@@ -170,7 +170,7 @@ var _ = SIGDescribe("[Feature:Windows] SecurityContext", func() {
 		framework.ExpectNotEqual(event, nil, "event should not be empty")
 		framework.Logf("Got event: %v", event)
 		expectedEventError := "container's runAsUserName (ContainerAdministrator) which will be regarded as root identity and will break non-root policy"
-		framework.ExpectEqual(true, strings.Contains(event.Message, expectedEventError), "Event error should indicate non-root policy caused container to not start")
+		gomega.Expect(strings.Contains(event.Message, expectedEventError)).To(gomega.BeTrue(), "Event error should indicate non-root policy caused container to not start")
```
Comment (carlory) [inline test/e2e/windows/security_context.go:173]:
```suggestion
		gomega.Expect(event.Message).Should(gomega.ContainsSubstring(expectedEventError), "Event error should indicate non-root policy caused container to not start")
```

File: test/e2e/windows/security_context.go [hunk 4]
```diff
@@ -188,7 +188,7 @@ var _ = SIGDescribe("[Feature:Windows] SecurityContext", func() {
 		framework.ExpectNotEqual(event, nil, "event should not be empty")
 		framework.Logf("Got event: %v", event)
 		expectedEventError := "container's runAsUserName (CONTAINERADMINISTRATOR) which will be regarded as root identity and will break non-root policy"
-		framework.ExpectEqual(true, strings.Contains(event.Message, expectedEventError), "Event error should indicate non-root policy caused container to not start")
+		gomega.Expect(strings.Contains(event.Message, expectedEventError)).To(gomega.BeTrue(), "Event error should indicate non-root policy caused container to not start")
```
Comment (carlory) [inline test/e2e/windows/security_context.go:191]:
```suggestion
		gomega.Expect(event.Message).Should(gomega.ContainsSubstring(expectedEventError), "Event error should indicate non-root policy caused container to not start")
```

**Review by pohly — APPROVED** [review]
/approve

**Pull request merged**